src/coin_test/data/datasets.py

```diff
@@ -80,6 +80,89 @@ def process(self, processors: list[Processor]) -> "Dataset":
+
+        Args:
+            dataset (Dataset): Dataset used to validate the parameters
+            timestamp (pd.Timestamp | None, optional): Timestamp to split on.
+            length (pd.Timedelta | None, optional): Timedelta from
+                beginning to split on.
+            percent (float | None, optional): Percentage of the dataset
+                between [0.0, 1.0] to split.
+
+        Returns:
+            pd.Timestamp | int: Timestamp or the index to split on
+
+        Raises:
+            ValueError: Must specify one and only one of timestamp, length, or percent
+        """
+        first_date = dataset.df.index[0].start_time  # type: ignore
+        last_date = dataset.df.index[-1].start_time  # type: ignore
+        if timestamp is None and length is None and percent is None:
+            raise ValueError("Must specify how to split dataset")
+        elif timestamp is not None and length is None and percent is None:
+            if first_date > timestamp or timestamp > last_date:
+                raise ValueError(
+                    f"Timestamp given({timestamp}) must be within dataset \
+                        bounds ({first_date}:{last_date})"
+                )
+            index = timestamp
+        elif timestamp is None and length is not None and percent is None:
+            split_timestamp = first_date + length
+
+            if first_date > split_timestamp or split_timestamp > last_date:
+                raise ValueError(
+                    f"Start time + Timeperiod({split_timestamp}) must be \
+                        within dataset bounds ({first_date}:{last_date})"
+                )
+            index = split_timestamp
+        elif timestamp is None and length is None and percent is not None:
+            if 0.0 >= percent or percent >= 1:
+                raise ValueError(
+                    f"Percentage({percent}) must be greater than 0 and less than 1"
+                )
+            index = int(dataset.df.shape[0] * percent)
+        else:
+            raise ValueError("Cannot specify multiple methods of splitting a dataset")
+
+        return index
+
+    def split(
+        self,
+        timestamp: pd.Timestamp | None = None,
+        length: pd.Timedelta | None = None,
+        percent: float | None = None,
+    ) -> tuple["Dataset", "Dataset"]:
+        """Split Dataset into Pre and Post split Datasets.
+
+        Args:
+            timestamp: [Optional] Timestamp to split Dataset on
+            length: [Optional] pd.Timedelta to specify length of the pre-split dataset
+            percent: [Optional] float percentage of the data to split on
+
+        Returns:
+            tuple: Pre and post split datasets of the same type as the original dataset
+        """
+        index = Dataset._calculate_split_index(self, timestamp, length, percent)
+        pre_df = self.df[:index]
+        post_df = self.df[index:].tail(-1)
```

Comment:
Do we want to force these to copy? As-is I think these are slices of original dataframe, and are still linked. E.g, modifying the pre or post df will modify the original one. Perhaps alright? Not sure.